src/terminal/parser/stateMachine.cpp

```diff
@@ -1885,14 +1891,23 @@ void StateMachine::ProcessString(const std::wstring_view string)
     // When we leave the loop, current has been advanced to the length of the string itself
     // (or one past the array index to the final char) so this `substr` operation doesn't +1
     // to include the final character (unlike the one inside the top of the loop above.)
-    _run = start < string.size() ? string.substr(start) : std::wstring_view{};
+    if (start < string.size())

```

Comment:
I don't quite understand this piece of comment about, be it before this PR or after. Still the logic here remains intact.

Reply:
It's saying that on line 1846, a `.substr` is called with the 2nd parameter having a `+1` on the end as a magic value that helps select the correct distance.

So one might quickly think that this similar `.substr` call on line 1888 should also have a `+1` magic value and there's an error. But then it goes on to clarify that the nature of the looping made the magic `+1` unnecessary.

Or at least that's my understanding of the comment. It looks like this code may have been revised a few times since it was originally written.